### PR TITLE
Reimplement Chart Repo without Umbrella Chart

### PR DESCRIPTION
**Pull request by goruha**
## What
* Refactor chart repo

## Why
* New version chartmuseum released

### CODE REVIEW

File: rootfs/conf/kops/helmfile.yaml [hunk 1]
```diff
@@ -58,20 +58,40 @@ releases:
...
+
+    # use amazon s3 storage as backend
+    - name: "env.open.STORAGE"
+      value: "amazon"
+
+    ### Required: STORAGE_AMAZON_BUCKET;
+    - name: "env.open.STORAGE_AMAZON_BUCKET"
+      value: '{{ env "CHARTMUSEUM_STORAGE_AMAZON_BUCKET" }}'
+
+    ### Required: STORAGE_AMAZON_PREFIX; e.g. /
```
Comment (aknysh) [inline rootfs/conf/kops/helmfile.yaml]: Required: CHARTMUSEUM_STORAGE_AMAZON_PREFIX

File: rootfs/conf/kops/helmfile.yaml [hunk 2]
```diff
@@ -58,20 +58,40 @@ releases:
...
+    - name: "env.open.STORAGE_AMAZON_PREFIX"
+      value: '{{ env "CHARTMUSEUM_STORAGE_AMAZON_PREFIX" }}'
+
+    ### Required: STORAGE_AMAZON_REGION; e.g. us-west-2
```
Comment (aknysh) [inline rootfs/conf/kops/helmfile.yaml]: Required: CHARTMUSEUM_STORAGE_AMAZON_REGION

File: rootfs/conf/kops/helmfile.yaml [hunk 3]
```diff
@@ -122,208 +127,118 @@ releases:
...
-    - name: "gateway.env.open.STORAGE"
-      value: '{{ coalesce (env "CHART_REPO_STORAGE") "amazon" }}'
+    # use amazon s3 storage as backend
+    - name: "env.open.STORAGE"
+      value: "amazon"
 
-    ### Required: CHART_REPO_STORAGE_AMAZON_BUCKET; e.g. cp-staging-chart-repo
-    - name: "gateway.env.open.STORAGE_AMAZON_BUCKET"
-      value: '{{ env "CHART_REPO_STORAGE_AMAZON_BUCKET" }}'
+    ### Required: STORAGE_AMAZON_BUCKET;
```
Comment (aknysh) [inline rootfs/conf/kops/helmfile.yaml]: CHARTMUSEUM_STORAGE_AMAZON_BUCKET

File: rootfs/conf/kops/helmfile.yaml [hunk 4]
```diff
@@ -122,208 +127,118 @@ releases:
...
+    - name: "env.open.STORAGE_AMAZON_BUCKET"
+      value: '{{ env "CHARTMUSEUM_STORAGE_AMAZON_BUCKET" }}'
 
-    ### Optional: CHART_REPO_STORAGE_AMAZON_PREFIX; e.g. /
-    - name: "gateway.env.open.STORAGE_AMAZON_PREFIX"
-      value: '{{ coalesce (env "CHART_REPO_STORAGE_AMAZON_PREFIX") "/" }}'
+    ### Required: STORAGE_AMAZON_PREFIX; e.g. /
```
Comment (aknysh) [inline rootfs/conf/kops/helmfile.yaml]: CHARTMUSEUM_STORAGE_AMAZON_PREFIX

File: rootfs/conf/kops/helmfile.yaml [hunk 5]
```diff
@@ -122,208 +127,118 @@ releases:
...
+    - name: "env.open.STORAGE_AMAZON_PREFIX"
+      value: '{{ env "CHARTMUSEUM_STORAGE_AMAZON_PREFIX" }}'
 
-    ### Optional: CHART_REPO_STORAGE_AMAZON_REGION; e.g. us-west-2
-    - name: "gateway.env.open.STORAGE_AMAZON_REGION"
-      value: '{{ coalesce (env "CHART_REPO_STORAGE_AMAZON_REGION") (env "AWS_REGION") }}'
+    ### Required: STORAGE_AMAZON_REGION; e.g. us-west-2
```
Comment (aknysh) [inline rootfs/conf/kops/helmfile.yaml]: CHARTMUSEUM_STORAGE_AMAZON_REGION

File: rootfs/conf/kops/helmfile.yaml [hunk 6]
```diff
@@ -122,208 +127,118 @@ releases:
...
     namespace: "kube-system"
     vendor: "kubernetes-helm"
-    default: "false"
-  chart: "cloudposse-incubator/chart-repo"
-  version: "0.2.1"
+    default: "true"
+  chart: "stable/chartmuseum"
+  version: "1.4.0"
   values:
-    - "values/chart-repo.yaml"
```
Comment (aknysh) [inline rootfs/conf/kops/helmfile.yaml]: was `chart-repo.yaml` renamed to `chartmuseum.yaml` in the repo?

Comment (goruha) [reply rootfs/conf/kops/helmfile.yaml]: No. There is no need in it.

**Comment by osterman** [comment]
What are the upgrade implications of this for our customers. I like the change, but we need to document the upgrade path.

File: rootfs/conf/kops/helmfile.yaml [hunk 7]
```diff
@@ -122,208 +127,121 @@ releases:
...
+    - name: "ingress.annotations.external-dns\\.alpha\\.kubernetes\\.io/target"
+      value: '{{ env "CHARTMUSEUM_INGRESS" }}'
 
-    ### Optional: CHART_REPO_SERVER_TTL; e.g. 60
+    # DNS TTL for CHARTMUSEUM_HOSTNAME; e.g. 60
     # FIXME: this is blocked by https://github.com/roboll/helmfile/issues/119
-    #- name: "server.ingress.annotations.external-dns\\.alpha\\.kubernetes\\.io/ttl"
-    #  value: {{ coalesce (env "CHART_REPO_SERVER_TTL") "60" }}
+    #- name: "ingress.annotations.external-dns\\.alpha\\.kubernetes\\.io/ttl"
+    #  value: "60"
```
[the remark this answers is not in the document]
Comment (osterman) [reply rootfs/conf/kops/helmfile.yaml]: The comments are inaccurate. Also, I would prefer we maintain the envs.

File: rootfs/conf/kops/helmfile.yaml [hunk 8]
```diff
@@ -58,20 +58,40 @@ releases:
...
     ### Required: CHARTMUSEUM_BASIC_AUTH_USER; e.g. server
     - name: "env.secret.BASIC_AUTH_USER"
       value: '{{ env "CHARTMUSEUM_BASIC_AUTH_USER" }}'
 
-    ### Required: CHARTMUSEUM_BASIC_AUTH_PASS; e.g. e.g. 15041434-3DC2-42F8-9358-F8AED8780A66
+    ### Required: CHARTMUSEUM_BASIC_AUTH_PASS; e.g. 15041434-3DC2-42F8-9358-F8AED8780A66
     - name: "env.secret.BASIC_AUTH_PASS"
       value: '{{ env "CHARTMUSEUM_BASIC_AUTH_PASS" }}'
 
+    ### Required: CHARTMUSEUM_IAM_ROLE;
```
Comment (osterman) [inline rootfs/conf/kops/helmfile.yaml]: follow convention. Use an `e.g.`

**Comment by osterman** [comment]
Document a migration path from what we had to what we have now. We'll need to carry this out with (2) customers.

1. What envs changed. 
2. What envs can be deleted.
3. What new envs need to be defined.

File: rootfs/conf/kops/helmfile.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (osterman) [reply rootfs/conf/kops/helmfile.yaml]: @goruha please review this comment. It was related to this: https://github.com/roboll/helmfile/issues/119

Namely, `helm` casts the integer to a string, but k8s expects an integer so it bombs.

[the remark this answers is not in the document]
Comment (osterman) [reply rootfs/conf/kops/helmfile.yaml]: Now, `helmfile` supports inline `values`, so maybe we should use that.

[the remark this answers is not in the document]
Comment (osterman) [reply rootfs/conf/kops/helmfile.yaml]: <--- user error. forgot this was commented out.

**Comment by daveyu** [comment]
These are the env variable changes I found:
```
############################
OLD ENVS
############################
CHARTMUSEUM_BASIC_AUTH_PASS
CHARTMUSEUM_BASIC_AUTH_USER
CHARTMUSEUM_HOSTNAME
CHARTMUSEUM_INGRESS
CHARTMUSEUM_STORAGE_AMAZON_REGION
CHART_REPO_GATEWAY_BASIC_AUTH_PASS
CHART_REPO_GATEWAY_BASIC_AUTH_USER
CHART_REPO_GATEWAY_HOSTNAME
CHART_REPO_GATEWAY_INGRESS
CHART_REPO_SERVER_BASIC_AUTH_PASS
CHART_REPO_SERVER_BASIC_AUTH_USER
CHART_REPO_SERVER_HOSTNAME
CHART_REPO_SERVER_INGRESS
CHART_REPO_STORAGE_AMAZON_BUCKET
CHART_REPO_STORAGE_AMAZON_REGION
CHART_REPO_STORAGE_AWS_IAM_ROLE

############################
ADD ENVS
############################
CHARTMUSEUM_API_BASIC_AUTH_PASS
CHARTMUSEUM_API_BASIC_AUTH_USER
CHARTMUSEUM_IAM_ROLE
CHARTMUSEUM_STORAGE_AMAZON_BUCKET
CHARTMUSEUM_STORAGE_AMAZON_PREFIX
CHARTMUSEUM_STORAGE_AMAZON_REGION

############################
DELETE ENVS
############################
CHART_REPO_GATEWAY_BASIC_AUTH_PASS
CHART_REPO_GATEWAY_BASIC_AUTH_USER
CHART_REPO_GATEWAY_HOSTNAME
CHART_REPO_GATEWAY_INGRESS
CHART_REPO_SERVER_BASIC_AUTH_PASS
CHART_REPO_SERVER_BASIC_AUTH_USER
CHART_REPO_SERVER_HOSTNAME
CHART_REPO_SERVER_INGRESS
CHART_REPO_STORAGE_AMAZON_BUCKET
CHART_REPO_STORAGE_AMAZON_REGION
CHART_REPO_STORAGE_AWS_IAM_ROLE
```

File: rootfs/conf/kops/helmfile.yaml [hunk 9]
```diff
@@ -107,10 +116,6 @@ releases:
     #- name: "ingress.annotations.kubernetes\\.io/tls-acme"
     #  value: "true"
 
-    ### Required: CHARTMUSEUM_HOSTNAME; e.g. e.g. charts.us-west-2.staging.cloudposse.org
```
Comment (osterman) [inline rootfs/conf/kops/helmfile.yaml]: @goruha this is required and `external-dns` won't work without it

File: rootfs/conf/kops/helmfile.yaml [hunk 10]
```diff
@@ -134,196 +143,105 @@ releases:
...
-# References:
-#   - https://github.com/cloudposse/charts/blob/master/incubator/chart-repo/values.yaml
-#   - https://github.com/kubernetes/charts/blob/master/stable/chartmuseum/values.yaml
-#
-- name: "charts"
+- name: "charts-api"
   namespace: "kube-system"
   labels:
-    chart: "chart-repo"
+    chart: "chartmuseum-api"
```
Comment (osterman) [inline rootfs/conf/kops/helmfile.yaml]: this should be the name of the helm chart which is `chartmuseum`

**Pull request merged**